Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverOnboarding/Status.hs

```diff
@@ -0,0 +1,793 @@
+    dateOfUpload :: UTCTime
+  }
+  deriving (Show, Eq, Generic, ToJSON, FromJSON, ToSchema)
+
+data DocumentStatusItem = DocumentStatusItem
+  { documentType :: DDVC.DocumentType,
+    verificationStatus :: ResponseStatus,
+    verificationMessage :: Maybe Text,
+    verificationUrl :: Maybe BaseUrl
+  }
+  deriving (Show, Eq, Generic, ToJSON, FromJSON, ToSchema)
+
+data DLDetails = DLDetails
+  { driverName :: Maybe Text,
+    driverLicenseNumber :: Text,
+    operatingCity :: Text,
+    driverDateOfBirth :: Maybe UTCTime,
+    classOfVehicles :: [Text],
+    imageId1 :: Text,
+    imageId2 :: Maybe (Text),
+    dateOfIssue :: Maybe UTCTime,
+    createdAt :: UTCTime
+  }
+  deriving (Show, Eq, Generic, ToJSON, FromJSON, ToSchema)
+
+data RCDetails = RCDetails
+  { vehicleRegistrationCertNumber :: Text,
+    imageId :: Text,
+    operatingCity :: Text,
+    dateOfRegistration :: Maybe UTCTime,
+    vehicleCategory :: Maybe Text,
+    airConditioned :: Maybe Bool,
+    vehicleManufacturer :: Maybe Text,
+    vehicleModel :: Maybe Text,
+    vehicleColor :: Maybe Text,
+    vehicleDoors :: Maybe Int,
+    vehicleSeatBelts :: Maybe Int,
+    vehicleModelYear :: Maybe Int,
+    oxygen :: Maybe Bool,
+    ventilator :: Maybe Bool,
+    createdAt :: UTCTime
+  }
+  deriving (Show, Eq, Generic, ToJSON, FromJSON, ToSchema)
+
+statusHandler' ::
+  Id DP.Person ->
+  DMOC.MerchantOperatingCity ->
+  DTC.TransporterConfig ->
+  Maybe Bool ->
+  Maybe Bool ->
+  Maybe Bool ->
+  Maybe DVC.VehicleCategory ->
+  Maybe DL.DriverLicense ->
+  Flow StatusRes'
+statusHandler' personId merchantOperatingCity transporterConfig makeSelfieAadhaarPanMandatory multipleRC prefillData onboardingVehicleCategory mDL = do
+  let merchantId = merchantOperatingCity.merchantId
+      merchantOpCityId = merchantOperatingCity.id
+  person <- runInReplica $ Person.findById personId >>= fromMaybeM (PersonNotFound personId.getId)
+  let language = fromMaybe merchantOperatingCity.language person.language
+
+  driverDocuments <- fetchDriverDocuments personId merchantOperatingCity transporterConfig language
+
+  vehicleDocumentsUnverified <- fetchVehicleDocuments personId merchantOperatingCity transporterConfig language
+
+  whenJust (onboardingVehicleCategory <|> (mDL >>= (.vehicleCategory))) $ \vehicleCategory -> do
+    documentVerificationConfigs <- CQDVC.findByMerchantOpCityIdAndCategory merchantOpCityId vehicleCategory
+    let mandatoryVehicleDocumentVerificationConfigs = filter (\config -> config.documentType `elem` vehicleDocumentTypes && config.isMandatory) documentVerificationConfigs
+    when (null mandatoryVehicleDocumentVerificationConfigs) $ do
+      allDriverDocsVerified <- Extra.allM (\doc -> checkIfDocumentValid merchantOpCityId doc.documentType vehicleCategory doc.verificationStatus makeSelfieAadhaarPanMandatory) driverDocuments
+      when (allDriverDocsVerified && transporterConfig.requiresOnboardingInspection /= Just True) $ do
```

Comment:
enable driver only in `postDriverOperatorRespondHubRequest` api after offline inspection if `requiresOnboardingInspection`